### ISSUE TITLE
iOS11 Crashing/losing data

### ISSUE THREAD

**Issue by ammichael** [ionic-team/cordova-plugin-wkwebview-engine#164]
Hi there,

I've been using this plugin for a while now, and it worked great. Thanks (:

As I'm testing out my app's update for iOS 11, I've bumped a problem. First it was crashing at launch, so updated the version (which was 1.1.1-dev) for the latest, 1.1.4-dev. Then the crashing was fixed, but all my data storage (which is basically pouchdb using websql) were gone. Reinstalling the old version of the plugin restore the data, but then I have the crash problem.

Here is the crash log for 1.1.1:

```
WebKit`WebKit::CallbackMap::put:
    0x1945271f0 <+0>:   sub    sp, sp, #0x50             ; =0x50 
    0x1945271f4 <+4>:   stp    x22, x21, [sp, #0x20]
    0x1945271f8 <+8>:   stp    x20, x19, [sp, #0x30]
    0x1945271fc <+12>:  stp    x29, x30, [sp, #0x40]
    0x194527200 <+16>:  add    x29, sp, #0x40            ; =0x40 
    0x194527204 <+20>:  mov    x19, x1
    0x194527208 <+24>:  mov    x20, x0
    0x19452720c <+28>:  mov    x21, x8
    0x194527210 <+32>:  bl     0x18c923c04               ; WTF::RunLoop::isMain()
    0x194527214 <+36>:  tbz    w0, #0x0, 0x194527270     ; <+128>
    0x194527218 <+40>:  ldr    x8, [x19]
    0x19452721c <+44>:  ldr    x8, [x8, #0x18]
    0x194527220 <+48>:  str    x8, [x21]
    0x194527224 <+52>:  add    x8, x8, #0x1              ; =0x1 
    0x194527228 <+56>:  cmp    x8, #0x1                  ; =0x1 
    0x19452722c <+60>:  b.ls   0x194527274               ; <+132>
    0x194527230 <+64>:  ldr    x8, [x20]
    0x194527234 <+68>:  cbz    x8, 0x194527248           ; <+88>
    0x194527238 <+72>:  mov    x0, x20
    0x19452723c <+76>:  mov    x1, x21
    0x194527240 <+80>:  bl     0x194526d70               ; WTF::KeyValuePair<unsigned long long, WTF::RefPtr<WebKit::CallbackBase> >* WTF::HashTable<unsigned long long, WTF::KeyValuePair<unsigned long long, WTF::RefPtr<WebKit::CallbackBase> >, WTF::KeyValuePairKeyExtractor<WTF::KeyValuePair<unsigned long long, WTF::RefPtr<WebKit::CallbackBase> > >, WTF::IntHash<unsigned long long>, WTF::HashMap<unsigned long long, WTF::RefPtr<WebKit::CallbackBase>, WTF::IntHash<unsigned long long>, WTF::HashTraits<unsigned long long>, WTF::HashTraits<WTF::RefPtr<WebKit::CallbackBase> > >::KeyValuePairTraits, WTF::HashTraits<unsigned long long> >::lookup<WTF::IdentityHashTranslator<WTF::HashMap<unsigned long long, WTF::RefPtr<WebKit::CallbackBase>, WTF::IntHash<unsigned long long>, WTF::HashTraits<unsigned long long>, WTF::HashTraits<WTF::RefPtr<WebKit::CallbackBase> > >::KeyValuePairTraits, WTF::IntHash<unsigned long long> >, unsigned long long>(unsigned long long const&)
    0x194527244 <+84>:  cbnz   x0, 0x194527278           ; <+136>
    0x194527248 <+88>:  add    x8, sp, #0x8              ; =0x8 
    0x19452724c <+92>:  mov    x0, x20
    0x194527250 <+96>:  mov    x1, x21
    0x194527254 <+100>: mov    x2, x19
    0x194527258 <+104>: bl     0x19452746c               ; WTF::HashTableAddResult<WTF::HashTableIterator<unsigned long long, WTF::KeyValuePair<unsigned long long, WTF::RefPtr<WebKit::CallbackBase> >, WTF::KeyValuePairKeyExtractor<WTF::KeyValuePair<unsigned long long, WTF::RefPtr<WebKit::CallbackBase> > >, WTF::IntHash<unsigned long long>, WTF::HashMap<unsigned long long, WTF::RefPtr<WebKit::CallbackBase>, WTF::IntHash<unsigned long long>, WTF::HashTraits<unsigned long long>, WTF::HashTraits<WTF::RefPtr<WebKit::CallbackBase> > >::KeyValuePairTraits, WTF::HashTraits<unsigned long long> > > WTF::HashMap<unsigned long long, WTF::RefPtr<WebKit::CallbackBase>, WTF::IntHash<unsigned long long>, WTF::HashTraits<unsigned long long>, WTF::HashTraits<WTF::RefPtr<WebKit::CallbackBase> > >::inlineSet<unsigned long long const&, WTF::Ref<WebKit::CallbackBase> >(unsigned long long const&&&, WTF::Ref<WebKit::CallbackBase>&&)
    0x19452725c <+108>: ldp    x29, x30, [sp, #0x40]
    0x194527260 <+112>: ldp    x20, x19, [sp, #0x30]
    0x194527264 <+116>: ldp    x22, x21, [sp, #0x20]
    0x194527268 <+120>: add    sp, sp, #0x50             ; =0x50 
    0x19452726c <+124>: ret    
->  0x194527270 <+128>: brk    #0
    0x194527274 <+132>: brk    #0
    0x194527278 <+136>: brk    #0
```

Basically I need to get the 1.1.1 to work with iOS 11, or the 1.1.4 to keep my data.

Any ideas?
Help?

Thanks!
